libs/tendermint/consensus/state.go

```diff
@@ -927,7 +927,14 @@ func (cs *State) enterPropose(height int64, round int) {
 			cs.Step))
 		return
 	}
-	cs.trc.Pin("Propose-%d", round)
+	address := cs.privValidatorPubKey.Address()
+
+	isBp := "n"
+	if cs.isProposer != nil && cs.isProposer(address) {
```

Comment:
isProposer := cs.isProposer(address)

and `isProposer` could be used by the following code.